libmproxy/web/app.py

```diff
@@ -176,6 +177,60 @@ def post(self, flow_id):
             raise APIError(400, r)
 
 
+class ViewPluginFlowContent(RequestHandler):
+    def get(self, flow_id, message, plugin_id):
+        message = getattr(self.flow, message)
+
+        if not message.content:
+            raise APIError(400, "No content.")
+
+        content_encoding = message.headers.get("Content-Encoding", None)
+        if content_encoding:
+            content_encoding = re.sub(r"[^\w]", "", content_encoding)
+            self.set_header("Content-Encoding", content_encoding)
+
+        original_cd = message.headers.get("Content-Disposition", None)
+        filename = None
+        if original_cd:
+            filename = re.search("filename=([\w\" \.\-\(\)]+)", original_cd)
+            if filename:
+                filename = filename.group(1)
+        if not filename:
+            filename = self.flow.request.path.split("?")[0].split("/")[-1]
+
+        filename = re.sub(r"[^\w\" \.\-\(\)]", "", filename)
+        cd = "attachment; filename={}".format(filename)
+        self.set_header("Content-Disposition", cd)
+        self.set_header("Content-Type", "application/text")
+        self.set_header("X-Content-Type-Options", "nosniff")
+        self.set_header("X-Frame-Options", "DENY")
+
+        try:
+            type, lines = contentviews.get_content_view(
+                contentviews.get(plugin_id),
+                message.content,
+                headers=message.headers
+            )
+        except ContentViewException:
+            s = "Content viewer failed: \n" + traceback.format_exc()
+            self.add_event(s, "debug")
+            type, lines = contentviews.get_content_view(
+                contentviews.get("Raw"),
+                message.content,
+                headers=message.headers
+            )
+
+        def get_text(line):
+            for (style, text) in line:
+                yield text
+
+        content = u"\r\n".join(
+            u"".join(get_text(line)) for line in lines
+        )
```

Comment:
I think we want to do this concatentation on the client side so that we can add colors later on. :smiley: